rust/ql/lib/codeql/rust/internal/TypeMention.qll

```diff
@@ -106,6 +131,13 @@ class TypeParamMention extends TypeMention, TypeParam {
   override Type resolveType() { result = TTypeParamTypeParameter(this) }
 }
 
+// Used to represent implicit associated type type arguments in traits.
```

Comment:
type type -> type

Reply:
It's not the easiest thing to read but the repeated type is intentional as it's supposed to be "associated type" type arguments. I've reworded it so that the two "type"s don't occur back-to-back.